backend/api/serializers/course_serializer.py

```diff
@@ -73,6 +90,7 @@ class Meta:
             "excerpt",
             "description",
             "faculty",
+            "faculty_id",
```

Comment:
Why does the course serializer now contain both the faculty and the faculty_id apart?

Reply:
These changes were to remove the CreateCourseSerializer and put all the logic in the CourseSerializer to make it cleaner. But by using the FacultySerializer and passing a faculty object in a POST request to create a course, this fails when there is already a faculty with that ID ( but that is always the case because you never make a faculty, you just pass it to link it to the course).
Using this faculty_id field, you can pass a faculty_id in a request and this will automatically link the corresponding faculty to the course ( btw this field is not displayed in a GET request, only used in creation).